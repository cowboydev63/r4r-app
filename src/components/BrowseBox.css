.browse-tiles__container.r4r-DEFAULT {
    /* display: flex;
    flex-direction: column;
    flex: 1; */
}

.browse-tiles__container.r4r-DEFAULT .browse__tile.r4r-DEFAULT {
    margin: 5px 0;
}

.browse-tiles__container.r4r-DEFAULT .r4r-spinner {
    display: block;
    margin: 40px;
    width: auto;
}