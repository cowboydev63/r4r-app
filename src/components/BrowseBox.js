import React from 'react';
import PropTypes from 'prop-types';
import BrowseTile from './BrowseTile';
import './BrowseBox.css';

class BrowseBox extends React.PureComponent {
    static propTypes = {
        facetFilters: PropTypes.array.isRequired,
        className: PropTypes.string,
        searchFunction: PropTypes.func.isRequired,
        filterType: PropTypes.string.isRequired,
    }

    static defaultProps = {
        facetFilters: [],
    }

    render() {
        return(
            <div className={`browse-tiles__container ${this.props.className ? `browse-tiles__container--${ this.props.className }` : '' }`}>
                {
                    this.props.facetFilters.map((filter, idx) => {
                        return (
                            <BrowseTile
                                key={ idx }
                                label={ filter.label }
                                onClick={ 
                                    this.props.searchFunction({
                                        filterType: this.props.filterType,
                                        filter: filter.key
                                    })
                                }
                            />
                        )
                    })
                }
            </div>
        )
    }
}

export default BrowseBox;