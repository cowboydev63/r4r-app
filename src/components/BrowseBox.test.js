import React from 'react';
import { shallow } from 'enzyme';
import BrowseBox from './BrowseBox';

describe('BrowseBox Component', () => {
  it('renders correctly', () => {
    const wrapper = mountWithTheme(<BrowseBox />);
    expect(wrapper).toMatchSnapshot();
  })

})
