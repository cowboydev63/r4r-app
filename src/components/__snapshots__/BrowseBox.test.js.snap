// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseBox Component renders correctly 1`] = `
<ThemeProvider
  theme={Object {}}
>
  <BrowseBox
    displayCount={false}
    facets={Object {}}
    filterType=""
    searchFunction={[Function]}
  >
    <Theme
      className="browse-tiles__container"
      element="div"
    >
      <div
        className="browse-tiles__container  "
      />
    </Theme>
  </BrowseBox>
</ThemeProvider>
`;
